source/extensions/filters/network/common/redis/client.h

```diff
@@ -41,6 +41,20 @@ class PoolCallbacks {
    * Called when a network/protocol error occurs and there is no response.
    */
   virtual void onFailure() PURE;
+
+  /**
+   * Called when a MOVED redirection error is received, and the request must be retried.
+   * @param value supplies the MOVED error response
+   * @return bool true if the request is successfully redirected, false otherwise
+   */
+  virtual bool onMovedRedirection(const Common::Redis::RespValue& value) PURE;
```

Comment:
AFAICT the implementation of these functions is identical. What's the difference? If there is no difference right now can we just merge into `onRedirection()` and split later if needed?

Reply:
the difference is very subtle right now, but will become more pronounced after Henry finishes some of his work. Right now, the ask redirection handler has to add "asking" to the command before retrying, while the moved redirection does not. In the future, ask redirection is transient, and will therefore be handled differently for cluster topology updates.

Reply:
Given how much code is duplicated, what about having a single function called `onRedirection(...)` that takes a `type` enum parameter? Then when the logic differs we can pivot later. I think that's a simpler approach which will make this change smaller.